pkg/provider/aws/resources/rds.go

```diff
@@ -0,0 +1,327 @@
+	instance.CredentialsFile = &core.RawFile{
+		FPath:   credsPath,
+		Content: credsBytes,
+	}
+	instance.CredentialsPath = credsPath
+	rdsPolicyDoc := CreateAllowPolicyDocument([]string{"rds-db:connect"}, []core.IaCValue{{Resource: instance, Property: RDS_CONNECTION_ARN_IAC_VALUE}})
+	rdsPolicy := NewIamPolicy(cfg.AppName, fmt.Sprintf("%s-connectionpolicy", orm.ID), orm.Provenance(), rdsPolicyDoc)
+	dag.AddDependency(rdsPolicy, instance)
+	dag.AddDependency(rdsPolicy, NewAccountId())
+	dag.AddDependency(rdsPolicy, NewRegion())
+
+	var proxy *RdsProxy
+	if proxyEnabled {
+		role := NewIamRole(cfg.AppName, fmt.Sprintf("%s-ormsecretrole", orm.ID), []core.AnnotationKey{orm.Provenance()}, RDS_ASSUME_ROLE_POLICY)
+		secret := NewSecret(orm.Provenance(), orm.Id(), cfg.AppName)
+
+		secretVersion := NewSecretVersion(secret, credsPath)
+		secretVersion.Type = "string"
+		secretPolicyDoc := CreateAllowPolicyDocument([]string{"secretsmanager:GetSecretValue"}, []core.IaCValue{{Resource: secret, Property: core.ARN_IAC_VALUE}})
+		secretPolicy := NewIamPolicy(cfg.AppName, fmt.Sprintf("%s-ormsecretpolicy", orm.ID), orm.AnnotationKey, secretPolicyDoc)
+		role.ManagedPolicies = append(role.ManagedPolicies, core.IaCValue{Resource: secretPolicy, Property: core.ARN_IAC_VALUE})
+		dag.AddDependency(secretPolicy, secret)
+
+		proxy = NewRdsProxy(orm, cfg.AppName, securityGroups, subnets, role, secret)
+		dag.AddDependency(proxy, secret)
+		proxyTargetGroup := NewRdsProxyTargetGroup(orm, cfg.AppName, instance, proxy)
+		dag.AddDependenciesReflect(secretVersion)
+		dag.AddDependenciesReflect(proxyTargetGroup)
+		dag.AddDependenciesReflect(proxy)
+		dag.AddDependenciesReflect(role)
+		dag.AddDependenciesReflect(secretPolicy)
+	}
+	dag.AddDependenciesReflect(instance)
+	dag.AddDependenciesReflect(subnetGroup)
+	return instance, proxy, nil
+}
+
+func (i *RdsInstance) GetConnectionPolicy(dag *core.ResourceGraph) *IamPolicy {
+	var pol *IamPolicy
+	upstreamDeps := dag.GetUpstreamDependencies(i)
+	for _, dep := range upstreamDeps {
+		if res, ok := dep.Source.(*IamPolicy); ok {
+			if len(res.Policy.Statement[0].Action) == 1 && res.Policy.Statement[0].Action[0] == "rds-db:connect" {
+				return res
+			}
+		}
+	}
+	return pol
+}
+
+// generateUsername generates a random username for the rds instance.
+//
+// The first letter of an RDS username must be a letter
+func generateUsername() string {
+	rand.Seed(time.Now().UnixNano())
+	initialChars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz")
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789")
+	length := 12
+	var b strings.Builder
+	b.WriteRune(chars[rand.Intn(len(initialChars))])
+	for i := 0; i < length-1; i++ {
+		b.WriteRune(chars[rand.Intn(len(chars))])
+	}
+	return b.String()
+}
+
+// generatePassword generates a random password for the rds instance.
+func generatePassword() string {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
from [`rand` godocs](https://pkg.go.dev/math/rand):

> For random numbers suitable for security-sensitive work, see the crypto/rand package

Reply:
are you suggesting we use another package? im not sure what more we would get out of that when were just generating a 12 char string

Reply:
they would need to know when klotho was run, not when the instance was created, which seems nearly impossible at the moment. But if its more secure, still good to switch it out so i can do that

Reply:
Run logs might not be hard to find, either. Think about at Amazon — anyone internally could tell you when pretty much any build job happened, company-wide.

I think the "fix" is pretty easy, too: we could just use crypto/rand to populate a []byte, and then base64-encode that.